src/core/ParticlePropertyIterator.hpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2010-2022 The ESPResSo project
+ *
+ * This file is part of ESPResSo.
+ *
+ * ESPResSo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ESPResSo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef ESPRESSO_SRC_CORE_PARTICLE_EXTRACT_PROPERTIES_HPP
+#define ESPRESSO_SRC_CORE_PARTICLE_EXTRACT_PROPERTIES_HPP
+
+#include "BoxGeometry.hpp"
+#include "Particle.hpp"
+#include "ParticleRange.hpp"
+#include "grid.hpp"
+#include <boost/iterator/transform_iterator.hpp>
+#include <boost/range/iterator_range.hpp>
+#include <tuple>
+#include <utils/Vector.hpp>
+
+namespace ParticlePropertyRange {
+
+namespace {
+template <class Kernel>
+auto create_transform_range(ParticleRange const &particles, Kernel kernel) {
+  auto transform_iterator_begin =
+      boost::make_transform_iterator(particles.begin(), kernel);
+  auto transform_iterator_end =
+      boost::make_transform_iterator(particles.end(), kernel);
+  return boost::make_iterator_range<decltype(transform_iterator_begin)>(
+      transform_iterator_begin, transform_iterator_end);
+}
+} // namespace
+
+auto pos_range(ParticleRange const &particles) {
+  auto return_pos = [](Particle &p) { return p.pos(); };
```

Comment:
Do we want ranges to yield particle properties by copy or by reference?
```suggestion
  auto return_pos = [](Particle &p) -> Utils::Vector3d & { return p.pos(); };
```

Reply:
Also, according to the code coverage report, this function is never called. Consider removing it.

Reply:
You are right, I forgot to remove this now redundant function